components/brave_rewards/browser/rewards_service.h

```diff
@@ -275,6 +275,13 @@ class RewardsService : public KeyedService {
                                      const std::string& query,
                                      ConnectExternalWalletCallback) = 0;
 
+  // Completes an external wallet login flow for the specified wallet provider
+  // using a collection of parameters returned from the provider's login page.
+  virtual void ConnectExternalWallet(
```

Comment:
For the 3.0 Rewards page, the parsing of the URL is performed on the front-end. This new overload allows the front-end to submit the parsed info to the engine, bypassing URL parsing.